### PR TITLE
add `clear` function to MapCache and memoized functions

### PR DESCRIPTION
**Pull request by ahamid**
re: #330

### CODE REVIEW

File: lodash.js [hunk 1]
```diff
@@ -8636,6 +8647,7 @@
         return result;
       };
       memoized.cache = new memoize.Cache;
+      memoized.clear = function() { memoized.cache.clear(); };
```
Comment (jdalton) [inline lodash.js]: No need for this line.

File: lodash.js [hunk 2]
```diff
@@ -13741,6 +13753,7 @@
     MapCache.prototype.get = mapGet;
     MapCache.prototype.has = mapHas;
     MapCache.prototype.set = mapSet;
+    MapCache.prototype.clear = mapClear;
```
Comment (jdalton) [inline lodash.js]: Move this above the `MapCache.prototype['delete']` assignment.

**Comment by jdalton** [comment]
Hi @ahamid! Have you signed [our CLA](http://dojofoundation.org/about/claForm)?

**Comment by ahamid** [comment]
Just signed the CLA, but feel free to close this unless you hear further demand, I just wanted to try my hand at this.

**Comment by jdalton** [comment]
Thanks!

**Pull request merged**